src/aspect.ts

```diff
@@ -1,27 +1,80 @@
 import { Handle } from './interfaces';
 import { createHandle } from './lang';
 
+/**
+ * An object that provides the necessary APIs to be MapLike
+ */
+export interface MapLike<K, V> {
+	get(key: K): V;
+	set(key: K, value?: V): this;
+}
+
+/**
+ * An internal type guard that determines if an value is MapLike or not
+ *
+ * @param value The value to guard against
+ */
+function isMapLike(value: any): value is MapLike<any, any> {
+	return value && typeof value.get === 'function' && typeof value.set === 'function';
+}
+
+export interface Indexable {
```

Comment:
We debated this before, regarding "should we have anything we can't `tslint`" (see: dojo/meta#43) and in my opinion, if that were true, then we would have unmaintainable code as there are always higher order and subjective things a linter can't enforce.

> We either need to code to our guidelines or update them.

I agree, I was trying to determine why you raised it to help determine if we needed to change them or not.  I guess it makes sense for exported interfaces to be alphabetically ordered at the top of the file, after the imports, where as other items, I still think make better sense close to their first usage site.